core/runtime/src/main/java/io/quarkus/runtime/util/ContainerRuntimeUtil.java

```diff
@@ -130,14 +128,13 @@ private static ContainerRuntime loadContainerRuntimeFromSystemProperty() {
             return null;
         }
 
-        ContainerRuntime containerRuntime = ContainerRuntime.valueOf(runtime);
-
-        if (containerRuntime == null) {
-            log.warnf("System property %s contains an unknown value %s. Ignoring it.",
-                    CONTAINER_RUNTIME_SYS_PROP, runtime);
+        try {
+            return ContainerRuntime.valueOf(runtime);
```

Comment:
valueOf won't return null, but it might throw an exception.  I tested the change:

```
 -Dquarkus-local-container-runtime=GUGU

[WARNING] [io.quarkus.runtime.util.ContainerRuntimeUtil]
    System property quarkus-local-container-runtime contains an unknown value GUGU.
    Ignoring it. java.lang.IllegalArgumentException:
    No enum constant io.quarkus.runtime.util.ContainerRuntimeUtil.ContainerRuntime.GUGU
```